### PR TITLE
Test some `QML` components

### PR DESCRIPTION
**Pull request by mohsenD98**
This pull request adds a set of tests for the QML components provided by the EditorWidgets module. The tests cover various aspects of each component, including their initial state, configuration options, and functionality. The tests are organized into separate functions, each of which focuses on a specific component.

The components tested in this pull request are:
  

**TextEdit**: Tests its ability to set and get values, as well as its configuration options for multiline and HTML text.
**Range:** Verifies its initial value, default values, and configuration options for style, precision, min, max, step, and suffix.
**DateTime:** Tests its display formats, calendar popup, and field format options.
**CheckBox:** Verifies its initial state, checked and unchecked states, and configuration options for display method and type.
**ValueMap:** Tests its properties, including toggle buttons and combobox models, as well as its value and current key value.
**UuidGenerator:** Verifies its initial state and value, as well as its isLoaded and isAdding properties.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f371acf3572fb351787ab99701be31f0e94440c3: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5375-36d4fffc62e7c2ab812a768947bc6bac4b854b2b-android-x86.apk)
</details>

File: test/test_qml.cpp [hunk 1]
```diff
@@ -157,11 +179,35 @@ class Setup : public QObject
       qmlInit( engine );
       engine->rootContext()->setContextProperty( QStringLiteral( "dataDir" ), mDataDir );
 
+
+      qmlRegisterType<QgsSnappingUtils>( "org.qgis", 1, 0, "SnappingUtils" );
+      qmlRegisterType<QgsMapLayerProxyModel>( "org.qgis", 1, 0, "MapLayerModel" );
+      qmlRegisterType<QgsVectorLayer>( "org.qgis", 1, 0, "VectorLayer" );
+      qmlRegisterType<QgsMapThemeCollection>( "org.qgis", 1, 0, "MapThemeCollection" );
+      // qmlRegisterType<QgsLocatorProxyModel>( "org.qgis", 1, 0, "QgsLocatorProxyModel" );
```
Comment (nirvn) [inline test/test_qml.cpp:187]: Remove?

File: test/test_qml.cpp [hunk 2]
```diff
@@ -27,7 +27,29 @@
...
 #include <qgsfeature.h>
 #include <qgsgeometry.h>
+#include <qgsmaplayerproxymodel.h>
+#include <qgsmapthemecollection.h>
 #include <qgspoint.h>
+#include <qgssnappingutils.h>
+#include <qgsvectorlayer.h>
+
+#include <valuemapmodel.h>
+// #include <QgsLocatorProxyMod
```
Comment (nirvn) [inline test/test_qml.cpp:37]: Remove?

File: test/qml/tst_editorwidgets.qml [hunk 3]
```diff
@@ -2,45 +2,356 @@ import QtQuick 2.3
...
+    range.value = 3
+    compare(range.widgetStyle, "TextField")
+    compare(range.precision, 1)
+    compare(range.min, -Infinity)
+    compare(range.max, Infinity)
+    compare(range.step, 1)
+    compare(range.suffix, "")
+
+    // Row
+    // compare(sliderRow.visible, true) // ERROR ? should work but not working!
```
Comment (nirvn) [inline test/qml/tst_editorwidgets.qml:147]: Can we remove for now until we can either fix the logic or fix what's wrong? :)

File: test/qml/tst_editorwidgets.qml [hunk 4]
```diff
@@ -2,45 +2,356 @@ import QtQuick 2.3
...
 
-    EditorWidgets.TextEdit {
-        id: textEdit
-        property var value: 'one'
-        property var config: undefined
-        property bool isEnabled: true
+    // TextField -> textField
```
Comment (nirvn) [inline test/qml/tst_editorwidgets.qml:149]: remove?

File: test/qml/tst_editorwidgets.qml [hunk 5]
```diff
@@ -2,45 +2,356 @@ import QtQuick 2.3
...
+
+    const testTimes = ["2023-01-01", "2023-01-01 23:33:56"]
+    const displayFormats = ["yyyy-MM-dd", "yyyy-MM.dd", "yyyy-MM-dd HH:mm:ss", "HH:mm:ss", "HH:mm"]
+    const results = ["2023-01-01", "2023-01.01", "2023-01-01 00:00:00", "00:00:00", "00:00",
+                     "2023-01-01", "2023-01.01", "2023-01-01 23:33:56", "23:33:56", "23:33"]
+    let resultIdx = 0
+
+    for (let time of testTimes) {
+      for (let format of displayFormats) {
+        dateTime.config = {
```
Comment (nirvn) [inline test/qml/tst_editorwidgets.qml:200]: Nice, I love how you're stress-testing values and formats.

File: test/qml/tst_editorwidgets.qml [hunk 6]
```diff
@@ -2,45 +2,356 @@ import QtQuick 2.3
...
+   * Test case for textEdit widget
+   *
+   * This function tests the textEdit widget's functionality by setting its config to allow multiline and use html,
+   * setting its value to "six", verifying that it contains "six", resetting its config to disable multiline and html,
+   * setting its value to "seven", and comparing its children's texts with "six" and "seven".
+   */
+  function test_01_textEdit() {
+    const textReadonlyValue = textEdit.children[0]
+    const textField = textEdit.children[1]
+    const textArea = textEdit.children[2]
```
Comment (nirvn) [inline test/qml/tst_editorwidgets.qml:77]: We should change this to something a bit more flexible, such as relying on objectName (and creating a small utility function to find / return an item by objectName). Not for this PR though.

**Review by nirvn — COMMENTED** [review]
@mohsenD98 , just a few comments, thanks, this already adds good additional CI checks. Lovely.

File: test/test_qml.cpp (hunk 1 above)
Comment (mohsenD98) [reply test/test_qml.cpp:187]: I', not sure how to register this one ?! its registered in `qgismobileapp.cpp` but couldn't find its `#include` signature.

File: test/qml/tst_editorwidgets.qml (hunk 6 above)
Comment (mohsenD98) [reply test/qml/tst_editorwidgets.qml:77]: Yes, i agree.

File: test/qml/tst_editorwidgets.qml (hunk 5 above)
Comment (mohsenD98) [reply test/qml/tst_editorwidgets.qml:200]: thanks

File: test/qml/tst_editorwidgets.qml (hunk 3 above)
Comment (mohsenD98) [reply test/qml/tst_editorwidgets.qml:147]: lets keep it to not forget, and comeback later ?

File: test/test_qml.cpp (hunk 1 above)
Comment (nirvn) [reply test/test_qml.cpp:187]: @mohsenD98 , you can use QGIS' excellent API documentation site to get the answer: https://api.qgis.org/api/classQgsLocatorProxyModel.html

At the top, you'll see the include, in this case: `#include <qgslocatormodel.h>`

Comment (mohsenD98) [reply test/test_qml.cpp]: thank you, fantastic :+1:

File: test/test_qml.cpp [hunk 7]
```diff
@@ -25,9 +25,30 @@
...
+#include <qgsquickmapsettings.h>
+#include <qgsquickmaptransform.h>
+#include <qgsrelationmanager.h>
+#include <qgssnappingutils.h>
+#include <qgsunittypes.h>
+#include <qgsvectorlayer.h>
+#include <qgsvectorlayereditbuffer.h>
+#include <qgswkbtypes.h>
+
+#include <valuemapmodel.h>
```
Comment (nirvn) [inline test/test_qml.cpp:51]: one last thing @mohsenD98, this is one of our own header, it should go  up alongside the #include "positioning.h" and not use <>.

**Review by nirvn — APPROVED** [review]

**Pull request merged**